provider/blob/src/blob_schema.rs

```diff
@@ -98,6 +118,90 @@ impl<'data> BlobSchemaV1<'data> {
     }
 }
 
+/// Version 2 of the ICU4X data blob schema.
+#[derive(Clone, Copy, Debug, serde::Deserialize, yoke::Yokeable)]
+#[yoke(prove_covariance_manually)]
+#[cfg_attr(feature = "export", derive(serde::Serialize))]
+pub(crate) struct BlobSchemaV2<'data> {
+    /// Map from key hash to locale trie.
+    /// Weak invariant: should be sorted.
+    #[serde(borrow)]
+    pub keys: &'data ZeroSlice<DataKeyHash>,
+    /// Map from locale to buffer index.
+    /// Weak invariant: the `usize` values are valid indices into `self.buffers`
+    /// Weak invariant: there is at least one value for every integer in 0..self.buffers.len()
+    /// Weak invariant: keys and locales are the same length
+    // TODO: Make ZeroTrieSimpleAscii<[u8]> work when in this position.
+    #[serde(borrow)]
+    pub locales: &'data VarZeroSlice<[u8]>,
+    /// Vector of buffers
+    #[serde(borrow)]
+    pub buffers: &'data VarZeroSlice<[u8], Index32>,
+}
+
+impl Default for BlobSchemaV2<'_> {
+    fn default() -> Self {
+        Self {
+            keys: ZeroSlice::new_empty(),
+            locales: VarZeroSlice::new_empty(),
+            buffers: VarZeroSlice::new_empty(),
+        }
+    }
+}
+
+impl<'data> BlobSchemaV2<'data> {
+    pub fn load(&self, key: DataKey, req: DataRequest) -> Result<&'data [u8], DataError> {
+        let idx0 = self
```

Comment:
Can we call this `trie_index` or something?

Reply:
done